### PR TITLE
add `--project` flag – fixes [#1527]

### LINKED ISSUE #1527: No monorepo support for cli lint
## Problem 

I can not use the --config anymore. The `package.json` and the `project.inlang.config` need to be on the same dir right now.

## Proposal
Add a --project option as we have it in the compile command of paraglide-js.

@inlang/cli

### PR DESCRIPTION
**Pull request by felixhaeberle**
fixes #1527

### CODE REVIEW

File: inlang/source-code/cli/src/commands/machine/translate.ts [hunk 1]
```diff
@@ -8,9 +8,10 @@ import prompts from "prompts"
 
 export const translate = new Command()
 	.command("translate")
+	.requiredOption("--project <path>", "Path to the project settings file.", "./project.inlang.json")
```
Comment (samuelstroschein) [inline inlang/source-code/cli/src/commands/machine/translate.ts:11]: what about a variable consumed by commands who need it to ensure consistency without relying on global options? 

```ts
export const projectOption = ["--project <path>", "Path to the project settings file", "./project.inlang.json"]
```

File: inlang/source-code/cli/src/utilities/getInlangProject.ts [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
 import { telemetry } from "../services/telemetry/implementation.js"
+import { normalizePath } from "@lix-js/fs"
 
 /**
  * Gets the inlang project and exists if the project contains errors.
  */
-export async function getInlangProject(): Promise<InlangProject> {
+export async function getInlangProject(args: { projectPath: string }): Promise<InlangProject> {
 	const baseDirectory = process.cwd()
-	const settingsFilePath = resolve(baseDirectory, "project.inlang.json")
```
Comment (samuelstroschein) [inline inlang/source-code/cli/src/utilities/getInlangProject.ts:11]: why is resolve not required anymore?

File: inlang/source-code/cli/src/commands/machine/translate.ts (hunk 1 above)
Comment (felixhaeberle) [reply inlang/source-code/cli/src/commands/machine/translate.ts]: done ✅

File: inlang/source-code/cli/src/utilities/getInlangProject.ts (hunk 2 above)
Comment (felixhaeberle) [reply inlang/source-code/cli/src/utilities/getInlangProject.ts:11]: removed it for normalizedPath, but let's keep resolve where we can.

File: inlang/source-code/cli/src/commands/lint/index.ts [hunk 3]
```diff
@@ -3,11 +3,12 @@ import Table from "cli-table3"
 import { getInlangProject } from "../../utilities/getInlangProject.js"
 import { log } from "../../utilities/log.js"
 import type { InlangProject, MessageLintReport } from "@inlang/sdk"
+import { projectOption } from "../../utilities/globalFlags.js"
 
 export const lint = new Command()
 	.command("lint")
 	.description("Commands for linting translations.")
-	.requiredOption("--project <path>", "Path to the project settings file.", "./project.inlang.json")
+	.requiredOption(projectOption.flags, projectOption.description, projectOption.defaultValue)
```
Comment (samuelstroschein) [inline inlang/source-code/cli/src/commands/lint/index.ts:11]: I think you can spread the project option

Comment (felixhaeberle) [reply inlang/source-code/cli/src/commands/lint/index.ts:11]: no somehow this raised type issues thus typed array.

**Pull request merged**